sample-resource-plugin/README.md

```diff
@@ -0,0 +1,187 @@
+# Resource Sharing and Access Control Plugin
+
+This plugin demonstrates resource sharing and access control functionality, providing sample resource APIs and marking it as a resource sharing plugin via resource-sharing-spi. The access control is implemented on Security plugin and will be performed under the hood.
+At present only admin and resource owners can modify/delete the resource
+
+## PreRequisites
+
+Publish SPI to local maven before proceeding:
+```shell
+./gradlew clean :opensearch-security-spi:publishToMavenLocal
+```
+
+System index feature must be enabled to prevent direct access to resource. Add the following setting in case it has not already been enabled.
+```yml
+plugins.security.system_indices.enabled: true
+```
+
+## Features
+
+- Create, update, get, delete SampleResource, as well as share and revoke access to a resource.
+
+## Installation
+
+1. Clone the repository:
+   ```bash
+   git clone git@github.com:opensearch-project/security.git
+   ```
+
+2. Navigate to the project directory:
+   ```bash
+   cd sample-resource-plugin
+   ```
+
+3. Build and deploy the plugin:
+   ```bash
+   $ ./gradlew clean build -x test -x integrationTest -x spotbugsIntegrationTest
+   $ ./bin/opensearch-plugin install file: <path-to-this-plugin>/sample-resource-plugin/build/distributions/opensearch-sample-resource-plugin-<version-qualifier>.zip
+   ```
+
+
+## User setup:
+1. **No Index-Level Permissions Required**
+    - **Resource access is controlled at the cluster level**.
+    - Users **do not** need explicit index-level permissions to access shared resources.
+
+2. **Sample Role Configurations**
+    - Below are **two sample roles** demonstrating how to configure permissions in `roles.yml`:
+
+    ```yaml
+    sample_full_access:
+     cluster_permissions:
+       - 'cluster:admin/sample-resource-plugin/*'
+
+    sample_read_access:
+     cluster_permissions:
+       - 'cluster:admin/sample-resource-plugin/get'
+    ```
+
+4. **Interaction Rules**
+    - If a **user is not the resource owner**, they must:
+        - Be assigned **a role with `sample_read_access`** permissions.
+        - **Have the resource shared with them** via the resource-sharing API.
+    - A user **without** the necessary `sample-resource-plugin` cluster permissions:
+        - **Cannot access the resource**, even if it is shared with them.
+    - A user **with `sample-resource-plugin` permissions** but **without a shared resource**:
+        - **Cannot access the resource**, since resource-level access control applies.
+
+
+## API Endpoints
+
+The plugin exposes the following six API endpoints:
+
+### 1. Create Resource
+- **Endpoint:** `POST /_plugins/sample_resource_sharing/create`
```

Comment:
nit: can we introduce or user a convention?

`/_plugins/_{name_of_plugin}`/api/{resource_type}/{action}`

Iin this case the action would be create.

Reply:
this is sample plugin's readme that pertains to actions specific to the sample plugin. This README aims to provide information about setup and testing the sample plugin. IMO it should be fine as is.

Reply:
I've made it verbose intentionally.